internal/shell/shell.go

```diff
@@ -76,17 +80,20 @@ func NewShell(config ShellConfig, db *db.Db) (*Shell, error) {
 }
 
 func (sh *Shell) Run() error {
-	err := sh.resetState()
-	if err != nil {
-		return err
-	}
 	defer sh.state.readline.Close()
 
 	if !sh.config.QuietMode {
 		fmt.Print(sh.getWelcomeMessage())
 	}
 
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt, syscall.SIGTERM)
+
 	for !sh.state.interruptReadEvalPrintLoop {
+		go func() {
+			<-signals
+			sh.CancelQuery()
+		}()
```

Comment:
here you're creating one goroutine for every user entry. This means that, if I have:
```
> select 1;
> select 1;
> select 1;
> selec^C
```
We have 4 goroutines creates for the same porpoises.

Also, this code aims to catch SIGTERM, and that's similar to what we do for non-interactive run (a.k.a `--exec` flag). So, it's interesting to add this logic on the same function level that we that we do for `--exec` flag. That is, if we have the logic inside `RunShellLine`. We should have the same logic for interactive shell inside `RunShell`

Reply:
Done, thanks!